### PR TITLE
WIP: Fix vector dataset

### PR DESCRIPTION
**Pull request by drroe**
Work in progress. This will be needed for cpptraj pull request: https://github.com/Amber-MD/cpptraj/pull/789

Right now this will build, but many tests will fail since the underlying data structures have changed for vector-related stuff. Essentially the previous DataSet_Vector class has been split into DataSet_Vector_XYZ (no origins) and DataSet_Vector_OXYZ (has origins). 

@hainm it appears that many tests with the main build are failing (see https://travis-ci.org/Amber-MD/pytraj); it may be tough to properly evaluate this PR while this is the case.

Any edits from @hainm that can help finish this PR will be very very welcome.

### CODE REVIEW

**Comment by hainm** [comment]
> @hainm it appears that many tests with the main build are failing (see https://travis-ci.org/Amber-MD/pytraj); it may be tough to properly evaluate this PR while this is the case.

Yeah, the linux tests have been failing due to MPI stuff. Please only look at the MacOS build. 

Please note that this PR always fails to build if cpptraj's PR is not merged yet. :D 

I can take over this PR once you're done with your initial changes. thanks Dan.

**Comment by hainm** [comment]
e.g: 

`
pytraj/analysis/c_action/actionlist.cpp:669:10: fatal error: 'DataSet_Vector_XYZ.h' file not found
#include "DataSet_Vector_XYZ.h"
         ^~~~~~~~~~~~~~~~~~~~~~
`

File: pytraj/all_actions.py [hunk 1]
```diff
@@ -794,7 +794,7 @@ def radgyr_tensor(traj=None,
     c_dslist, _ = do_action(traj, command, c_action.Action_Radgyr)
     k0, v0 = c_dslist[0].key, c_dslist[
         0].values.copy()  # use copy to avoid early memory free
-    k1, v1 = c_dslist[1].key, c_dslist[1].possible_data6
+    k1, v1 = c_dslist[1].key, c_dslist[1].data
```
Comment (hainm) [inline pytraj/all_actions.py]: nice :D

**Comment by drroe** [comment]
I also see a lot of seemingly spurious failures that seem to occur because of different output formatting:
```
File "/u/droe/.local/lib/python3.8/site-packages/pytraj-2.0.5.dev0-py3.8-linux-x86_64.egg/pytraj/analysis/c_action/actionlist.cpython-38-x86_64-linux-gnu.so", line ?, in pytraj.analysis.c_action.actionlist.compute
Failed example:
    data['myrms']
Expected:
    array([ 3.46476336,  3.4343108 ,  2.94523273, ...,  4.21848857,
            4.4566457 ,  3.94477017])
Got:
    array([3.46476336, 3.4343108 , 2.94523273, ..., 4.21848857, 4.4566457 ,
           3.94477017])
```
Any ideas how to fix that? (This is the master branch btw)

**Comment by drroe** [comment]
BTW, this is Ubuntu 18.04.3 LTS, anaconda 3 (python 3.8)

**Comment by hainm** [comment]
> I also see a lot of seemingly spurious failures that seem to occur because of different output formatting:

You can ignore it, I can do the cleanup later. Thanks.

**Comment by hainm** [comment]
I will reply to other comments later.

**Comment by hainm** [comment]
> Yeah, I just mean when I'm testing on my local machine. Right now it actually just hangs at either test_parallel_pmap/test_pmap.py::TestCpptrajCommandStyle::test_reference or test_parallel_pmap/test_pmap.py::TestParallelMapForHbond::test_pmap_hbond_with_solvent_bridge forever - not sure why. Is there a way to skip the tests that shouldn't run?

This is the command we are using in jenkins so you can try it. 
```
cd tests && pytest -vs --ignore=test_parallel_pmap
```

**Comment by hainm** [comment]
@drroe Please apply attached patch (`git apply <...>`) to fix most of the errors.

[patch.pytraj.txt](https://github.com/Amber-MD/pytraj/files/4094027/patch.pytraj.txt)

There are still 2 outstanding errors and I think you know better than me since they relate to your changes. :d

```
test_analysis/test_radgyr.py::test_radgyr_tensor FAILED
test_analysis/test_vector_analysis.py::TestVectorAnalysisModule::test_ired_vector FAILED
```

PS: To run a single test, just
```
cd tests
pytest -vs test_analysis/test_radgyr.py::test_radgyr_tensor
```

**Comment by drroe** [comment]
Note I'm pausing this PR - decided to hold off on the changes in cpptraj. I'll keep this here for now just in case.

**Comment by hainm** [comment]
@drroe What's the status of this PR? I guess you do not want to follow this path anymore, right?

**Comment by drroe** [comment]
> What's the status of this PR? I guess you do not want to follow this path anymore, right?

Ultimately I think the benefits of this do not come close to outweighing the headache that would be involved in making this work. I'll close it now; we can always open it back up if needed.

**Pull request closed without merging**